keps/sig-node/4112-passdown-resources-to-cri/README.md

```diff
@@ -0,0 +1,958 @@
+
+The UpdateContainerResourcesRequest message is extended to pass down unmodified
+resource requests from the PodSpec.
+
+```diff
+ message UpdateContainerResourcesRequest {
+     // ID of the container to update.
+     string container_id = 1;
+     // Resource configuration specific to Linux containers.
+     LinuxContainerResources linux = 2;
+     // Resource configuration specific to Windows containers.
+     WindowsContainerResources windows = 3;
+     // Unstructured key-value map holding arbitrary additional information for
+     // container resources updating. This can be used for specifying experimental
+     // resources to update or other options to use when updating the container.
+     map<string, string> annotations = 4;
++
++    // Kubernetes resource spec of the container
++    ContainerResourceConfig container_resources = 5;
+ }
+```
+
+#### PodSandboxConfig
+
+The PodSandboxConfig message (part of the RunPodSandbox request) will be
+extended to contain information about resources of all its containers known at
+the pod creation time. The container resources here are non-binding and only
+informational, e.g. for the runtime to prepare for optimal allocation of
+resources of all the containers of the Pod.
+
+```diff
+ message PodSandboxConfig {
+ 
+ ...
+ 
+     // Optional configurations specific to Linux hosts.
+     LinuxPodSandboxConfig linux = 8;
+     // Optional configurations specific to Windows hosts.
+     WindowsPodSandboxConfig windows = 9;
++
++    // Kubernetes resource spec of the containers in the pod.
++    PodResourceConfig pod_resources = 10;
+ }
+ 
++message PodResourceConfig {
++    repeated ContainerResourceConfig init_containers = 1;
++    repeated ContainerResourceConfig containers = 2;
++}
+```
+
+### kubelet
+
+Kubelet will be be extended to pass down the unmodified resource requests and
+limits to the container runtime in all related CRI requests, i.e.
+RunPodSandbox, CreateContainer and UpdateContainerResources.
+
+For example, take a PodSpec:
+
+```yaml
+apiVersion: v1
+kind: Pod
+...
+spec:
+  containers:
+    - name: cnt-1
+      ...
+      resources:
+        requests:
+          cpu: 1
+          memory: 1G
+        limits:
+          cpu: 2
+          memory: 2G
+    - name: cnt-2
+      ...
+      resources:
+        requests:
+          cpu: 100m
+          memory: 100M
+          vendor.com/xpu: 1
+        limits:
+          cpu: 200m
+          memory: 200M
+          vendor.com/xpu: 1
+```
+
+Then kubelet will send the following RunPodSandboxRequest when creating the Pod
+(represented here in yaml format):
+
+```yaml
+RunPodSandboxRequest:
+  config:
+  ...
+    podResources:
+      containers:
+        - name: cnt-1
+          requests:
+            cpu: 1
+            memory: 1G
+          limits:
+           cpu: 2
+           memory: 2G
+        - name: cnt-2
+          requests:
+            cpu: 100m
+            memory: 100M
+            vendor.com/xpu: 1
```

Comment:
For clarification: This does not indicate the properties of the resource that was actually allocated for a container requesting one of these devices?

Reply:
That's very much true. I think I'll add a note about this in the KEP somewhere

Reply:
ping @elezar may I close this as resolved?